### PR TITLE
Improve liveness for 'STORE_BLK(lcl_var)'.

### PR DESCRIPTION
**Pull request by sandreenko**
The rule for an untracked variable was weaker so converting a tracked variable to untracked sometimes gave us better code.

Don't set `lvMustInit ` for tracked `lclVars` that are compiler temps if they don't contain GC fields.

<details>
  <summary>Tiny SPMI improvements on master but I need it for my future changes.</summary>
  

```
tests.pmi.windows.x64.checked.mch
3 total methods with Code Size differences (3 improved, 0 regressed), 0 unchanged.
Top method improvements (bytes):
          -7 (-3.61% of base) : 247502.dasm - TailCallOptTest:Caller2(System.Object,long,double,TypedDouble,TwoInts,TypedDouble):bool
          -7 (-3.37% of base) : 247500.dasm - TailCallOptTest:Caller1(System.Object,long,double,TypedDouble,TypedDouble,TypedDouble):bool
          -7 (-3.66% of base) : 249520.dasm - Program:Caller(int,int,int,int,MyStruct):int

libraries.pmi.windows.x64.checked.mch
1 total files with Code Size differences (1 improved, 0 regressed), 0 unchanged.
          -7 (-13.21% of base) : 7338.dasm - Microsoft.FSharp.Text.StructuredPrintfImpl.LayoutOps:boundedUnfoldL(Microsoft.FSharp.Core.FSharpFunc`2[__Canon,__Canon],Microsoft.FSharp.Core.FSharpFunc`2[Nullable`1,__Canon],Microsoft.FSharp.Core.FSharpFunc`2[Nullable`1,Boolean],System.Nullable`1[Int32],int):Microsoft.FSharp.Collections.FSharpList`1[[Microsoft.FSharp.Text.StructuredPrintfImpl.Layout, FSharp.Core, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]]
```

</details>

Diff example:
[before.txt](https://github.com/dotnet/runtime/files/6046989/before.txt)
[after.txt](https://github.com/dotnet/runtime/files/6046990/after.txt)

### CODE REVIEW

**Comment by sandreenko** [comment]
@erozenfeld, could you please review it if you have time? It is not what I asked about but a different issue, the one that I asked I could not have fixed and will leave it for now.
PTAL @dotnet/jit-contrib,

**Comment by sandreenko** [comment]
I have changed the solution to better analyse "store_blk" and use "store_lcl_var" in the cases where "store_blk" was used with no reason.

It gives improvements on the same set of methods, so the original commit was dropped, maybe I will still need in the future, with additional struct changes, but for now, it does not produce diffs.

The new diffs:
```
benchmarks.run.windows.x64.checked.mch
1 total methods with Code Size differences (1 improved, 0 regressed), 0 unchanged.
libraries.pmi.windows.x64.checked.mch
12 total methods with Code Size differences (12 improved, 0 regressed), 0 unchanged.
tests.pmi.windows.x64.checked.mch
3 total methods with Code Size differences (3 improved, 0 regressed), 0 unchanged
```

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -6264,6 +6264,15 @@ GenTree* Compiler::gtNewOneConNode(var_types type)
...
 
+GenTreeLclVar* Compiler::gtNewStoreLclVar(unsigned dstLclNum, GenTree* src)
+{
+    GenTreeLclVar* store = new (this, GT_STORE_LCL_VAR) GenTreeLclVar(GT_STORE_LCL_VAR, src->TypeGet(), dstLclNum);
+    store->gtOp1         = src;
+    store->gtFlags       = (src->gtFlags & GTF_COMMON_MASK);
+    store->gtFlags |= GTF_VAR_DEF | GTF_ASG;
+    return store;
+}
+
```
Comment (briansull) [inline src/coreclr/jit/gentree.cpp:6275]: Nice improvement

File: src/coreclr/jit/liveness.cpp [hunk 2]
```diff
@@ -405,6 +406,10 @@ void Compiler::fgPerNodeLocalVarLiveness(GenTree* tree)
                         // states separately.
                         byrefStatesMatchGcHeapStates = false;
                     }
+                    if (entire)
+                    {
+                        fgMarkUseDef(dummyLclVarTree);
+                    }
```
Comment (briansull) [inline src/coreclr/jit/liveness.cpp:412]: This is good too.

**Review by briansull — APPROVED** [review]
Looks Good

**Review by erozenfeld — COMMENTED** [review]
LGTM

**Pull request merged**